### ISSUE TITLE
Virtual paging for sources that don't respect the "images per page" setting

### ISSUE THREAD

**Issue by Bionus** [Bionus/imgbrd-grabber#1412]
On some sources (or some APIs), it is not possible to set the number of images per page. An example is the Zerochan RSS API, which always return 100 images.

If the "images per page" setting is set to 20 for example, Grabber could simply load the first Zerochan page, and use it for the 5 first "virtual pages", each containing 20 images.

⚠️ Some sources return exotic images count (37, 42, etc.), so it might not always be divisible by the images per page setting. In those cases, Grabber might need to load two pages to satisfy the setting.

If the user's "images per page" setting is higher than the number of images returned, Grabber might also need to load more pages, until there are enough images.

**Example with 100 images returned and 20 images per page**

* 1: page 1, 1-20
* 2: page 1, 21-40
* 3: page 1, 41-60
* 4: page 1, 61-80
* 5: page 1, 81-100
* 6: page 2, 1-20
* ...

**Example with 37 images returned and 20 images per page**

* 1: page 1, 1-20
* 2: page 1, 21-37 + page 2, 1-3
* 3: page 2, 4-23
* ...

**Example with 10 images returned and 20 images per page**

* 1: page 1, 1-10 + page 2, 1-10
* 2: page 3, 1-10 + page 4, 1-10
* ...

**Comment by MrAndre96** [comment]
Yeah this would be a good idea. Come across this multipele times when I use certain sources

**Comment by brazenvoid** [comment]
There can be further optional uses for such a feature. Like a setting can make Grabber enforce the image per page setting after post processing. Which will definitely reduce the amount of clicks involved and make the search experience seamless. Not like now that if one has an extensive blacklist, one can end up with much lower amount of images per page.
